lib/portlayer/exec/commit.go

```diff
@@ -246,6 +216,88 @@ func Commit(op trace.Operation, sess *session.Session, h *Handle, waitTime *int3
 	return nil
 }
 
+// batchBlockOnFunc is a batching routine that batch processes incoming requests.
+// Incoming request signals the batching routine by throwing into a batch channel. When this routine is performing operation on
+// the previous requests, new requests will be batched in the channel, waiting for next iteration to process.
+func batchBlockOnFunc(ctx context.Context, batch chan chan error, operation func(operation trace.Operation) error) {
+	op := trace.FromContext(ctx, "Add container VM to VMGroup dispatch routine")
+
+	for {
+		members := make([]chan error, 0, 5) // batching queue
+		var req chan error
+		var ok bool
+
+		// block and wait for first request
+		select {
+		case req, ok = <-batch:
+			if !ok {
+				return
+			}
+			if req == nil {
+				continue
+			}
+			members = append(members, req)
+		case <-op.Done(): // when parent context is cancelled, quit
+			return
+		}
+
+		// fetch batched requests
+		for len(batch) > 0 {
+			req, ok = <-batch
+			if !ok {
+				return // batch channel closed, quit
```

Comment:
I think we should actually drop this return condition here - we've already dequeued one operation to be batched at a minimum. I think we should dispatch that then use the return condition in the initial `select` to exit the routine.